weave

```diff
@@ -1009,7 +1009,7 @@ with_container_addresses() {
 }
 
 echo_addresses() {
-    echo $1 $3 $4
+    echo $(printf "%.12s" $1) $3 $4
```

Comment:
In the case that `$3` and/or `$4` are blank, this will output extra spaces compared to the original.

And the smoke-test checks for exact string equality.